--- InputDevice.java.org	2011-01-15 16:39:59.000000000 -0800
+++ InputDevice.java	2011-01-08 17:21:39.000000000 -0800
@@ -21,6 +21,7 @@
 import android.view.MotionEvent;
 import android.view.Surface;
 import android.view.WindowManagerPolicy;
+import android.os.SystemProperties;
 
 import java.io.PrintWriter;
 import java.io.FileInputStream;
@@ -48,6 +49,7 @@
     /** Number of jumpy points to drop for touchscreens that need it. */
     private static final int JUMPY_TRANSITION_DROPS = 3;
     private static final int JUMPY_DROP_LIMIT = 3;
+    private static final int HWROTATION = SystemProperties.getInt("ro.sf.hwrotation", 0) / 90;
 
     final int id;
     final int classes;
@@ -750,6 +752,7 @@
                 long curTimeNano, Display display, int orientation,
                 int metaState) {
             boolean isMouse = (device.classes & RawInputEvent.CLASS_MOUSE) != 0;
+            orientation = (orientation + HWROTATION) % 4;
             if (mSkipLastPointers) {
                 mSkipLastPointers = false;
                 mLastNumPointers = 0;
