--- PhoneUtils.java.org	2011-01-15 16:40:59.000000000 -0800
+++ PhoneUtils.java	2011-01-15 03:53:05.000000000 -0800
@@ -2355,8 +2358,10 @@
     private static void activateSpeakerIfDocked(Phone phone) {
         if (DBG) log("activateSpeakerIfDocked()...");
 
-        if (PhoneApp.mDockState == Intent.EXTRA_DOCK_STATE_DESK ||
-                PhoneApp.mDockState == Intent.EXTRA_DOCK_STATE_CAR) {
+        //if (PhoneApp.mDockState == Intent.EXTRA_DOCK_STATE_DESK ||
+        //        PhoneApp.mDockState == Intent.EXTRA_DOCK_STATE_CAR) {
+        // galaxy tab has no Earpiece so always activate speaker
+        {
             if (DBG) log("activateSpeakerIfDocked(): Phone in a dock -> may need to turn on speaker.");
             PhoneApp app = PhoneApp.getInstance();
             BluetoothHandsfree bthf = app.getBluetoothHandsfree();
